modules/internal/ChapelIO.chpl

```diff
@@ -426,18 +783,42 @@ module ChapelIO {
   // these are overridden to not be inout
   // since they don't change when read anyway
   // and it's much more convenient to be able to do e.g.
-  //   reader & new ioLiteral("=")
+  //   reader <~> new ioLiteral("=")
+
+  /* Overlead to support reading an ioLiteral without
+     passing ioLiterals by reference, so that
+
+     .. code-block:: chapel
+
+       reader <~> new ioLiteral("=")
+
+     works without requiring an explicit temporary value to store
+     the ioLiteral.
+   */
   inline proc <~>(r: Reader, lit:ioLiteral):Reader {
     var litCopy = lit;
     r.readwrite(litCopy);
     return r;
   }
+  /* Overlead to support reading an ioNewline without
+     passing ioNewline by reference, so that
+
+     .. code-block:: chapel
+
+       reader <~> new ioNewline("=")
+
+     works without requiring an explicit temporary value to store
+     the ioNewline.
+   */
   inline proc <~>(r: Reader, nl:ioNewline):Reader {
     var nlCopy = nl;
     r.readwrite(nlCopy);
     return r;
   }
   
+  /* Explicit call for reading a literal as an
+     alternative to using ioLiteral.
```

Comment:
Please include a link for ioLiteral here, too.  (and the ioNewline references that follow)